.github/workflows/update-flake-lock.yml

```
@@ -0,0 +1,29 @@
+name: update-flake-lock
+on:
+  workflow_dispatch: # allows manual triggering
+  schedule:
+    - cron: '0 0 * * 1,4' # Run twice a week
+
+jobs:
+  lockfile:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Nix
+        uses: cachix/install-nix-action@v19
+        with:
+          extra_nix_config: |
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+      # `bors merge` will automerge if tests succeed
+      # Requires this github to be installed: https://app.bors.tech/
+      - name: Update flake.lock
+        uses: DeterminateSystems/update-flake-lock@v16
```

```suggestion
        uses: DeterminateSystems/update-flake-lock@a0c5484d59e7d32ccfb416fe8bae8db583b9febb # v16
```
I recently learned that using revision instead of tags here prevents an attacker who gained control over the included repo to inject arbitrary code. What do you think?

---

I am wondering if dependabot can still upgrade those hashes

---

It can and it will even update the version comment behind the line, according to an issue I read.